Automattic/wp-calypso
# launchpad preview domain upsell

Fixes https://github.com/Automattic/wp-calypso/issues/73324 

alt approach to https://github.com/Automattic/wp-calypso/pull/73429

## Proposed Changes

Shows the domain callout upsell above the launchpad site preview by:
* Extracting the domain callout components visual content for reusability
* Sending this content down from the launchpad preview to be rendered by a new optional prop in the WebPreview we are using.

![Screenshot 2023-02-17 at 12 22 16 PM](https://user-images.githubusercontent.com/28742426/219721486-8c43ee31-5e14-4432-8a59-9042b38a73b4.png)

## Testing Instructions



* Test a launchpad site with free plan.
* visit the launchpad and ensure the banner is visible and works.
* smoke test other preview interaction and visuals across different device previews and screen sizes.
* veryify a `calypso_launchpad_preview_domain_upsell_impression` event fires on load of the upsell, and that `calypso_launchpad_preview_domain_upsell_click` fires when clicking.
* verify clicking the link in the upsell takes us to domains selection.
* upgrade to a plan (or test on a launchpad site with a paid plan)
* verify the upsell is gone and preview works as expected

smoke test other instances of the WebPreview such as theme previews from the theme showcase, or pattern assembler flow in (create new site -> write flow -> select a design -> scroll to bottom for create your own).

smoke test other instances of the upsell callout, such as in the page editor on a free plan:
![Screenshot 2023-02-17 at 12 21 42 PM](https://user-images.githubusercontent.com/28742426/219721254-b3b11937-743e-4ddd-9f9e-7bc26b616d13.png)


## Pre-merge Checklist
